registration/app/registration/controllers/Main.scala

```diff
@@ -30,17 +30,27 @@ final class Main @Inject()(notificationRegistrarSupport: RegistrarSupport, topic
   def register(lastKnownDeviceId: String): Action[Registration] = Action.async(BodyJson[Registration]) { request =>
     val registration = request.body
 
-    def registerWith(registrar: NotificationRegistrar): Future[Result] =
-      topicValidator.removeInvalid(registration.topics) flatMap {
-        case \/-(validTopics) => registrar
-          .register(lastKnownDeviceId, registration.copy(topics = validTopics))
-          .map { processResponse }
-        case -\/(e) => Future.successful(InternalServerError(e.reason))
-      }
+    def validate(topics: Set[Topic]) = {
+      topicValidator
+        .removeInvalid(topics)
+        .onSuccess {
+          case \/-(validTopics) =>
+            logger.debug(s"Successfully validated topics in registration (${registration.deviceId}), topics valid: [$validTopics]")
+          case -\/(e) =>
+            logger.error(s"Could not validate topics ${e.topicsQueried} for registration (${registration.deviceId}), reason: ${e.reason}")
+        }
+    }
+
+    def registerWith(registrar: NotificationRegistrar) =
+      registrar
+        .register(lastKnownDeviceId, registration)
+        .map {processResponse}
 
     registrarFor(registration) match {
+      case \/-(registrar) =>
+        validate(registration.topics)
```

Comment:
yes, problems are logged inside method and we don't want to chain it with registration flow